src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -52,6 +54,8 @@
 
 final class MonoSendMany<I, O> extends MonoSend<I, O> implements Scannable {
 
+	static final String KEY_ON_DISCARD = "reactor.onDiscard.local";
```

Comment:
Can we name this `reactor.netty.onDiscard.local`

Reply:
I guess not. What I need here is access to package-private reactor discard hook since in case of fusion This operator delegates discarding from the queue to the upstream. It means that for example https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/UnicastProcessor.java#L503 unicastProcessor.clear will be called and it will be discarding elements. In order to ensure their release, we have to intercept that and mimic `onDiscard` acceptor.

As you can see, we delegate discarding to the downstream subscriber. Here is the question which we should think about - is there anybody downstream to handle discarded elements? Should we that onDiscard propagation to downstream at all like at the https://github.com/reactor/reactor-netty/pull/1172/files#diff-34c9e1ba9859e522329366ebbb5ffb56R379. I mean who can be interested in released `ByteBuf`? If we remove that further delegation we can use standard functionality and create a static context with onDiscard hook to offer it to the upstream.

Reply:
So at some point if this `reactor.onDiscard.local` is changed/removed, the functionality will stop working. (this is not a public API)

Reply:
Correct. One of the options is to hack it is:

```java

static final Object ON_DISCARD_HOOK;

static {
	Context c = reactor.core.publisher.Operators#enableOnDiscard(null, (v) -> {});
    
    Map.Entry<Object,Object> entry = c.stream().findFirst().orElse(null);

    if (entry != null) {
       ON_DISCARD_HOOK = entry.getKey();
    }
    else {
       ON_DISCARD_HOOK = null;
    }
}
```

Reply:
Cc / @simonbasle